File: algorithm/leetcode/nim_game.cpp
#include<iostream>

bool canWinNim(int n) {
    if (n % 4 == 0){
        return false;
    }
    else{
        return true;
    }
}

int main(){
    std::cout << canWinNim(1) << std::endl;
    std::cout << canWinNim(2) << std::endl;
    std::cout << canWinNim(3) << std::endl;
    std::cout << canWinNim(4) << std::endl;
    std::cout << canWinNim(5) << std::endl;
    std::cout << canWinNim(6) << std::endl;
    std::cout << canWinNim(7) << std::endl;
    std::cout << canWinNim(8) << std::endl;
}

File: algorithm/leetcode/nim_game.go
package main

import (
	"fmt"
)

func canWinNim(n int) bool {
	if n%4 == 0 {
		return false
	} else {
		return true
	}
}

func main() {
	fmt.Println(canWinNim(1))
	fmt.Println(canWinNim(2))
	fmt.Println(canWinNim(3))
	fmt.Println(canWinNim(4))
	fmt.Println(canWinNim(5))
	fmt.Println(canWinNim(6))
	fmt.Println(canWinNim(7))
}
